File: proj2/calc.js
// "node calc.js op int int"
// args[2] = op
// args[3] = int
// args[4] = int

function do_op (op, i1, i2) {
  switch (op) {
    case "add":
      return [i1 + i2, "+"];
    case "sub":
      return [i1 - i2, "-"];
    case "mul":
      return [i1 * i2, "*"];
    case "div":
      return [i1 / i2, "/"];
    case "gt":
      if (i1 > i2) return [true, ">"];
      else return [false, ">"];
    case "ge":
      if (i1 >= i2) return [true, ">="];
      else return [false, ">="];
    case "lt":
      if (i1 < i2) return [true, "<"];
      else return [false, "<"];
    case "le":
      if (i1 <= i2) return [true, "<="];
      else return [false, "<="];
    case "eq":
      if (i1 == i2) return [true, "="];
      else return [false, "="];
    case "neq":
      if (i1 != i2) return [true, "!="];
      else return [false, "!="];
    default:
      throw "ERROR: The operation is invalid";
  }
  
}

const args = process.argv;

op = args[2];
i1 = parseInt(args[3]);
i2 = parseInt(args[4]);

[result, op] = do_op(op, i1, i2);

console.log(`${i1} ${op} ${i2} = ${result}`);

File: proj1/calc.py
# Calculator that works through terminal
# Should be run as "python calc.py op int int"
# Operations accepted - add, sub, mul, div
# This calculator also does comparisons between the two ints given in command line
# Comparisons accepted - gt, ge, le, lt, eq, neq
import sys

def do_op(op, i1, i2):
  if op == 'add':
    return i1 + i2, '+'
  elif op == 'sub':
    return i1 - i2, '-'
  elif op == 'mul':
    return i1 * i2, '*'
  elif op == 'div':
    return i1 / i2, '/'
  elif op == 'gt':
    if i1 > i2:
      return True, '>'
    else:
      return False, '>'
  elif op == 'ge':
    if i1 >= i2:
      return True, '>='
    else:
      return False, '>='
  elif op == 'lt':
    if i1 < i2:
      return True, '<'
    else:
      return False, '<'
  elif op == 'le':
    if i1 <= i2:
      return True, '<='
    else:
      return False, '<='
  elif op == 'eq':
    if i1 == i2:
      return True, '='
    else:
      return False, '='
  elif op == 'neq':
    if i1 != i2:
      return True, '!='
    else:
      return False, '!='

def main():
  op = sys.argv[1]
  i1 = int(sys.argv[2])
  i2 = int(sys.argv[3])

  if op not in ['add', 'sub', 'mul', 'div', 'gt', 'ge', 'le', 'lt', 'eq', 'neq']:
    print("ERROR: The operation argument is invalid.")
    sys.exit(1)

  result, op = do_op(op, i1, i2)

  print(f"{i1} {op} {i2} = {result}")

if __name__ == "__main__":
  main()